### PR TITLE
Improve webaudio player

### PR DESCRIPTION
**Pull request by joggienl**
- Add pause functionality to the player
- Improve some esdoc
- Remove async/await from the load function
- Added some helper functions

### CODE REVIEW

File: src/WebAudio/Player.js [hunk 1]
```diff
@@ -117,15 +154,60 @@ export default class Player extends AudioContext {
...
+   * Pause playback of audio.
+   */
   pause() {
     if (!this.audioBufferExists() && !this.audioSourceExists()) {
       return;
     }
 
+    const elapsed = this.audioContext.currentTime - this.startedAt;
     this.audioSource.stop();
+    this.pausedAt = elapsed;
```
Comment (kvdb) [inline src/WebAudio/Player.js]: perhaps explain this a bit more. i'd say: pausedAt = this.audioContext.currentTime

File: src/WebAudio/Player.js [hunk 2]
```diff
@@ -117,15 +154,60 @@ export default class Player extends AudioContext {
...
+   * Get the currentTime for the audio that is loaded.
+   *
+   * @returns {number} - The currentTime value.
+   */
+  getCurrentTime() {
+    if (this.pausedAt) {
+      return this.pausedAt;
+    }
+
+    if (this.startedAt) {
```
Comment (kvdb) [inline src/WebAudio/Player.js]: same here

File: src/WebAudio/Player.js (hunk 1 above)
Comment (joggienl) [reply src/WebAudio/Player.js]: I'm not sure what you mean?
this.audioContext.currentTime holds the current time of the audioContext. It is not related to the audio playback, that is how AudioContext works.

In other words, pausing the playback doesn't pause the AudioContext.

Source: https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/currentTime

File: src/WebAudio/Player.js (hunk 2 above)
Comment (joggienl) [reply src/WebAudio/Player.js]: I guess my same comment from above counts here as well.

File: src/WebAudio/Player.js (hunk 1 above)
Comment (joggienl) [reply src/WebAudio/Player.js]: "It is not related to the audio playback": it is of course related, but you will have to calculate it against the moment you started audio playback.

**Comment by joggienl** [comment]
Thanks!

**Pull request merged**